presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -95,6 +95,7 @@
     private double memoryRevokingTarget = 0.5;
     private double memoryRevokingThreshold = 0.9;
     private boolean parseDecimalLiteralsAsDouble;
+    private boolean useMarkDistinct = true;
```

Comment:
maybe make this false, for all of the tests to run

Reply:
Maybe later. I need to test it to make sure it's generally better than the current alternative. I added a test suite to verify various combinations of DISTINCT aggregations with and without this optimizer.